pom.xml

```diff
@@ -1039,8 +1039,8 @@ from system library in Java 11+ -->
                         </configuration>
                         <dependencies>
                             <dependency>
-                                <groupId>mysql</groupId>
-                                <artifactId>mysql-connector-java</artifactId>
+                                <groupId>com.mysql</groupId>
+                                <artifactId>mysql-connector-j</artifactId>
                                 <version>${mysql.version}</version>
                             </dependency>
```

Comment:
Is this dependency still required, or can it be removed?

Reply:
Why should the support of MySQL be removed while executing the flyway execution? So far as I know MySQL is still supported. If MySQL should not be supported anymore then the support for MySQL should be removed in a separate pull request.

Reply:
Entschuldigung, aber ich weiß einfach nicht, ob es üblich ist, Abhängigkeiten, die bereits an anderer Stelle in pom.xml stehen, bei den Plugins nochmals aufzuführen. Allerdings habe ich ausprobiert, was passiert, wenn ich die Zeilen oben einfach entferne. Ergebnis: `mvn dependency:analyze` ist weiterhin zufrieden, und alle Tests (`mvn clean install -B '-Pall-tests,flyway,checkstyle,!development'`, also inklusive Datenbankmigration mit flyway) laufen unverändert durch . Ich kenne also momentan keinen Anwendungsfall, für den diese Abhängigkeit hier konfiguriert sein muss.

Reply:
Dies ist eine Laufzeit-Abhängigkeit, die FlyWay zwingend benötigt, wenn FlyWay mit einem Oracle MySQL Server kommunizieren soll. Wenn im Hintergrund jedoch ein MariaDB basierender MySQL Server läuft, wird die andere angegebene Abhängigkeit automatisch genutzt. Die Abhängigkeit zu MariaDB wird auch dann benutzt, wenn in der genutzten URI diese mit `mariadb` beginnt statt mit `mysql`. Diese hier definierten Abhängigkeiten haben nichts mit der Anwendung selbst zu tun. Auch ein `mvn dependency:analyze` sagt nichts, da dies, wie schon geschrieben eine Abhängigkeit bei der Ausführung von FlyWay ist.

Reply:
Danke für die ausführliche Erklärung.

Reply:
Die Frage, ob neben dem verbreiteten MariaDB auch noch MySQL unterstützt werden muss (und falls ja wie lange), kann das Release-Management (@solth) vielleicht besser beantworten. Aber das ist, wie oben schon gesagt, ein eigenes Thema.

Reply:
Ich werde nicht sagen, dass Oracle MySQL nicht mehr unterstützt werden sollte. Der aktuelle Aufwand 2 MySQL-basierende Datenbank-Management-Systeme zu unterstützen, ist sehr gering. Es werden auch nicht die Features von MariaDB oder MySQL genutzt, die nur in einem DBMS vorhanden sind und man in dem anderen vergeblich sucht. Auch ist Oracle MySQL nicht tot und wird von Oracle stark weiter entwickelt.

Die Installationsanleitungen und auch die CI auf GitHub mit dem Ubuntu System dafür als Basis liefert standardmäßig Oracle MySQL aus und kein MariaDB. Ob sich dies mit neueren Ubuntu Versionen geändert hat, weiß ich nicht, da ich kein Ubuntu einsetze. Auch ist ansonsten der Verbreitungsgrad von Oracle MySQL nach wie vor sehr hoch.